externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetTables.scala

```diff
@@ -38,49 +39,33 @@ class GetTables(
     try {
       val tableEnv = sessionContext.getExecutionContext.getTableEnvironment
 
-      var catalogName = catalog
-      if (catalog == null || catalog.isEmpty) {
-        catalogName = tableEnv.getCurrentCatalog
-      }
-
-      val schemaPattern = toJavaRegex(schema).r.pattern
-      val tableNamePattern = toJavaRegex(tableName).r.pattern
+      val catalogName = if (StringUtils.isEmpty(catalog)) tableEnv.getCurrentCatalog else catalog
 
-      var tables = List[String]()
+      val schemaPattern = toJavaRegex(schema).r
+      val tableNamePattern = toJavaRegex(tableName).r
 
-      val optional = tableEnv.getCatalog(catalogName)
-      if (optional.isPresent) {
-        val currCatalog = optional.get()
-        tables = currCatalog.listDatabases().asScala
-          .filter(database =>
-            schemaPattern.matcher(database).matches())
-          .flatMap { database =>
-            currCatalog.listTables(database).asScala
-              .filter(identifier =>
-                tableNamePattern.matcher(identifier).matches())
-              .filter(identifier => {
-                // only table or view
-                if (!tableTypes.contains(Constants.TABLE_TYPE) || !tableTypes.contains(
-                    Constants.VIEW_TYPE)) {
-                  // try to get table kind
-                  Try(currCatalog.getTable(ObjectIdentifier.of(
-                    catalogName,
-                    database,
-                    identifier).toObjectPath)) match {
-                    case Success(table) => tableTypes.contains(table.getTableKind.name())
+      var tables = tableEnv.getCatalog(catalogName).asScala.toSeq.flatMap { flinkCatalog =>
+        flinkCatalog.listDatabases().asScala
+          .filter { case schemaPattern(_*) => true; case _ => false }
+          .flatMap { _schema =>
+            flinkCatalog.listTables(_schema).asScala
```

Comment:
In flink, it uses `database` to represent `schema`, e.g. `listTables(database)`. But, on the other side, it's ok to follow general understood. I am ok.